.github/workflows/build.yml

```diff
@@ -6,58 +6,28 @@ name: Run build and tests
 on:
   push:
     branches:
-      - '*'
+      - '**'
   pull_request:
     branches: [ master ]
 
 jobs:
   build:
+    permissions:
+      checks: write
+      pull-requests: write
+      contents: read
     environment: CIRelease
     runs-on: ubuntu-latest
-
     steps:
-      - name: Set up JDK 11
-        uses: actions/setup-java@v3
-        with:
-          java-version: 11
-          distribution: temurin
-      - uses: actions/checkout@v2
-        with:
-          fetch-depth: 0
-      - run: git fetch --all --tags
-
-      - name: Create .gradle dir
-        run: mkdir -p $HOME/.gradle
-      - name: Install gpg secret key
-        env:
-          SIGNINGFILE: ${{ secrets.SIGNINGFILE }}
-        run: |
-          cat <(echo -e "${{ secrets.SIGNINGFILE }}") | gpg --batch --import
-          gpg --list-secret-keys --keyid-format LONG
-      - name: Export gpg file
-        env:
-          SIGNINGKEYID: ${{ secrets.SIGNINGKEYID }}
-          SIGNINGPASSWORD: ${{ secrets.SIGNINGPASSWORD }}
-        run: |
-          gpg --batch --passphrase="$SIGNINGPASSWORD" --pinentry-mode loopback --export-secret-keys $SIGNINGKEYID > $HOME/.gradle/secrets.gpg
-      - name: Create gradle sproperties
-        env:
-          APIKEY: ${{ secrets.APIKEY }}
-          APISECRET: ${{ secrets.APISECRET }}
-          SIGNINGKEYID: ${{ secrets.SIGNINGKEYID }}
-          SIGNINGPASSWORD: ${{ secrets.SIGNINGPASSWORD }}
-          SONATYPEPASSWORD: ${{ secrets.SONATYPEPASSWORD }}
-          SONATYPEUSER: ${{ secrets.SONATYPEUSER }}
-        run: echo -e "gradle.publish.key=$APIKEY\ngradle.publish.secret=$APISECRET\nsigning.keyId=$SIGNINGKEYID\nsigning.password=$SIGNINGPASSWORD\nsigning.secretKeyRingFile=$HOME/.gradle/secrets.gpg\nsonatypeUsername=$SONATYPEUSER\nsonatypePassword=$SONATYPEPASSWORD" > $HOME/.gradle/gradle.properties
-      - name: Build and test with Gradle
-        env:
-          JAVA_OPTS: "-Xmx1024M -XX:MaxPermSize=512M -XX:ReservedCodeCacheSize=512M"
-          GRADLE_OPTS: "-Dorg.gradle.daemon=true"
-        run: ./gradlew test build :publishIntershopMvnPublicationToMavenRepository -s --scan
-      - name: Publish Unit Test Results
-        uses: EnricoMi/publish-unit-test-result-action/composite@v2
-        if: always()
+      - name: Setup
+        uses: IntershopCommunicationsAG/.github/actions/gradle-plugin-setup@v0.3
         with:
-          files: build/test-results/**/*.xml
-      - name: Post Build
-        run: rm -f $HOME/.gradle/gradle.properties && rm -f $HOME/.docker/config.json
+          signing-file: ${{ secrets.SIGNINGFILE }}
+          signing-key-id: ${{ secrets.SIGNINGKEYID }}
+          signing-password: ${{ secrets.SIGNINGPASSWORD }}
+          api-key: ${{ secrets.APIKEY }}
+          api-secret: ${{ secrets.APISECRET }}
```

Comment:
I discussed with @DevDavido that we want to remove the reliance on secrets from the build pipeline (and only have them in release) to make our actions compatible with Dependabot,

Reply:
We can do that. But publish without signing does not work. So I don't see a possibility to do that.

Reply:
The signing and publishing would only be performed in the release workflow.